### ISSUE TITLE
Incorrect notification that merges two posts together when a user favourites a post

### ISSUE THREAD

**Issue by mattwelke** [elk-zone/elk#3023]
When someone favourited a post, the notification I received appears to have merged the post with another and displayed it as a single post that the user favourited:

![Image](https://github.com/user-attachments/assets/54de3e13-3705-44f5-808b-c2868714ed3f)

These are the two posts:

![Image](https://github.com/user-attachments/assets/3d84a00e-8ca3-4657-afda-85de3a7bf94b)

![Image](https://github.com/user-attachments/assets/bf33be0b-8e20-4aad-9634-1a8b3aff4fed)

You can see how the user did not actually favourite the second post. They only favourited the first:

![Image](https://github.com/user-attachments/assets/360e5542-c294-46ba-9b0d-0f0d8924437f)

There was no GitHub template when I opened this issue, so I don't know which other information I should attach for debugging purposes. Feel free to ping me if you want me to attach more!

**Comment by shuuji3** [comment]
Hi, thanks for reporting the issue. Is this happening on the stable version of Elk (https://elk.zone)? If so, we recently fixed the notification issue by #3014, which was deployed as a Canary version (https://main.elk.zone). Could you try the Canary version to see if the same issue has been solved?

**Comment by mattwelke** [comment]
@shuuji3 It was actually tricky to reproduce. I found that once another notification came in, it seemed to kick things back into being correct. This is the version that is running right now on the instance;

```
A nimble Mastodon web client

Version

canary (f8b4f70@main)

Built

July 16, 2024 at 12:47:05 AM
```

if you want, we can close this and assume it won't happen again. If it does, we can re-open it. If I'm the one posting again in the future, I'll make sure to read the reproduction steps in those linked tickets and try to put together a reproduction.
